Cleanup Helm values

clintyoshimura opened the pull request:
<!--
Thanks for contributing!

We've designed this PR template to speed up the PR review and merge process - please use it.
-->

## Is there a related GitHub Issue?
#2650

## What is this change about?
Cleanup the Helm values to make them more understandable for operators. The api.builderName parameter does not seem to be used and the yaml that Helm generated was the same after it was removed. The api.lifecycle.stagingRequirements parameter was moved to global.stagingRequirements because both api and controllers use those parameters.

## Does this PR introduce a breaking change?
no

## Acceptance Steps
All tests pass

## Tag your pair, your PM, and/or team
@acosta11 

## Things to remember
- Include any links to related PRs, issues, stories, slack discussions, etc... that will help establish context.
- Is there anything else of note that the reviewers should know about this change?
- This project follows the Cloud Foundry [Code of Conduct](https://www.cloudfoundry.org/code-of-conduct/)
-->

clintyoshimura commented:
In addition to the moved parameters, there were a few other issues that were not as significant that could use discussion.
adminUserName - This is a top level parameter. Should it be under global? All the other top level parameters are groups.
api.expose - Expose the API component via Contour. Should Contour be part of the parameter name?

georgethebeatle commented:
`global` is a special top level group that is visible to all subcharts in case your helm chart has subcharts. Here are the relevant [docs](https://helm.sh/docs/chart_template_guide/subcharts_and_globals/#global-chart-values). We used to have subcharts in the past and global is a historical leftover. It would be better if we put all common values such as `stagingRequirements` to the top level. We should also move the values under `global` a level up so that they are top level values too.

georgethebeatle reviewed (changes requested):
See above comment

clintyoshimura commented:
@georgethebeatle We moved all the global helm values to the top-level and removed the global group. There will be korifi-ci changes required so when the PR is approved, we can push those changes to get the PR checks to pass and merge it that way. Everyone will have to rebase after that.
